tests/testthat/test-aggregate_alignment_loanbook_exposure.R

```diff
@@ -68,47 +68,178 @@ test_that("number of identified companies equals unique list of companies in inp
   n_companies_input_net <- length(unique(test_data_aggregate_alignment_loanbook_exposure_net$name_abcd))
 
   expect_equal(
-    test_output_aggregate_alignment_loanbook_exposure_net$n_companies,
+    test_output_aggregate_alignment_loanbook_exposure_net$aggregate$n_companies,
     n_companies_input_net
   )
 })
 
-test_that("number of identified companies equals unique list of companies in input data", {
-  n_companies_input_buildout <- test_data_aggregate_alignment_loanbook_exposure_bopo %>%
-    dplyr::filter(.data$direction == "buildout") %>%
+test_that("number of identified companies per direction equals unique list of companies in input data", {
+  n_companies_input_bopo <- test_data_aggregate_alignment_loanbook_exposure_bopo %>%
     dplyr::distinct(.data$name_abcd) %>%
     nrow()
 
-  n_output_buildout <- test_output_aggregate_alignment_loanbook_exposure_bopo %>%
+  n_output_buildout <- test_output_aggregate_alignment_loanbook_exposure_bopo$aggregate %>%
     dplyr::filter(.data$direction == "buildout") %>%
     dplyr::pull(.data$n_companies)
 
   expect_equal(
     n_output_buildout,
-    n_companies_input_buildout
+    n_companies_input_bopo
   )
 
-  n_companies_input_phaseout <- test_data_aggregate_alignment_loanbook_exposure_bopo %>%
-    dplyr::filter(.data$direction == "phaseout") %>%
-    dplyr::distinct(.data$name_abcd) %>%
-    nrow()
-
-  n_output_phaseout <- test_output_aggregate_alignment_loanbook_exposure_bopo %>%
+  n_output_phaseout <- test_output_aggregate_alignment_loanbook_exposure_bopo$aggregate %>%
     dplyr::filter(.data$direction == "phaseout") %>%
     dplyr::pull(.data$n_companies)
 
   expect_equal(
     n_output_phaseout,
-    n_companies_input_phaseout
+    n_companies_input_bopo
   )
 })
 
 test_that("net aggregate results have the same columns as buildout/phaseout aggregate results, plus three columns on loan exposure", {
   exposure_columns <- c("sum_loan_size_outstanding", "sum_exposure_companies_aligned", "share_exposure_aligned")
 
   expect_equal(
-    c(names(test_output_aggregate_alignment_loanbook_exposure_bopo), exposure_columns),
-    names(test_output_aggregate_alignment_loanbook_exposure_net)
+    c(names(test_output_aggregate_alignment_loanbook_exposure_bopo$aggregate), exposure_columns),
+    names(test_output_aggregate_alignment_loanbook_exposure_net$aggregate)
   )
 })
+
+# When an additional variable is passed via group_var, add group_var to
+# variables considered in aggregation (GH: 33)
```

Comment:
I think the tests are out of sync? You mention `group_var`, but I believe this has been superseded by the `.by` argument? (this happens in many places below).

Unless I misunderstand?

Reply:
No, you very much spotted that correctly